src/rpc/blockchain.cpp

```diff
@@ -118,11 +118,13 @@ UniValue blockheaderToJSON(const CBlockIndex* blockindex)
     result.pushKV("finalsaplingroot", blockindex->hashFinalSaplingRoot.GetHex());
     result.pushKV("time", (int64_t)blockindex->nTime);
     result.pushKV("nonce", blockindex->nNonce.GetHex());
-    result.pushKV("solution", HexStr(blockindex->nSolution));
     result.pushKV("bits", strprintf("%08x", blockindex->nBits));
     result.pushKV("difficulty", GetDifficulty(blockindex));
     result.pushKV("chainwork", blockindex->nChainWork.GetHex());
 
+    CBlockHeader header = blockindex->GetBlockHeader();
+    result.pushKV("solution", HexStr(header.nSolution));
```

Comment:
Nit: this changes the order of the fields in the returned JSON, which is unnecessary.